### ISSUE TITLE
API Vault Files issue: 404 not found error if filename contains space by get method

### ISSUE THREAD

**Issue by laye0619** [coddingtonbear/obsidian-local-rest-api#138]
By using version 2.5.4, everything works well.

However, after upgraded to version 3, I ran into a problem that
- if filename contains space, then I got 404 not found
- if filename contains Chinese, then I got 404 not found

For example,

it works file when

`
curl -X 'GET' \
  'http://192.168.1.23:27123/vault/07.abc/giffgaff.md' \
  -H 'accept: application/vnd.olrapi.note+json' \
  -H 'Authorization: Bearer HIDE_HERE'
`

I got 404 not found when
`
curl -X 'GET' \
  'http://192.168.1.23:27123/vault/07.abc/my plan.md' \
  -H 'accept: application/vnd.olrapi.note+json' \
  -H 'Authorization: Bearer HIDE_HERE'
`

or 

`
curl -X 'GET' \
  'http://192.168.1.23:27123/vault/07.abc/my%20plan.md' \
  -H 'accept: application/vnd.olrapi.note+json' \
  -H 'Authorization: Bearer HIDE_HERE'
`

or

`
curl -X 'GET' \
  'http://192.168.1.23:27123/vault/07.abc/中文字符.md' \
  -H 'accept: application/vnd.olrapi.note+json' \
  -H 'Authorization: Bearer HIDE_HERE'
`

**Comment by esodesod** [comment]
I can confirm similair issues after upgrading Local Rest API to 3.0.2. I've noticed the "note recall" feature now suddenly throws "ERR" (in Obsidian Web), if I have a note with matching url in YAML Front Matter for `url: ` (example below) where note names **contains** spacing (which most of my notes do). If I remove spacing in the file name (e.g. changing title from `test test` to `test`, it works - using same content within the note.

**Note:** It does not however seem to affect the recall feature if the url is within the note body/content (e.g. not in YAML Front Matter), using exactly the same note and URL, just moving the URL from YAML Front Matter to the note body, then recall feature works again.

Example, **not working**
Filename: `test test.md`
```md
---
url: https://github.com/coddingtonbear/obsidian-local-rest-api/issues
---

test

```

**Note:** If renaming file to `test.md`, it works (like expected)

Moving content into body also **works** (no error)
Filename: `test test.md`

```md
---
author: test
---

url: https://github.com/coddingtonbear/obsidian-local-rest-api/issues

```

**Comment by coddingtonbear** [comment]
Whoops -- I definitely see the bug I created.  One of the modules that were upgraded between 2.x and 3.x required that I change some aspects of how requests are routed, and I hadn't noticed that a side-effect of those changes was that I'd now need to manually URLDecode the URL.  I've fixed this via 8700c11 & released the updated version as 3.0.3.

Sorry about that!
